tests/testSpanSets.cc

```diff
@@ -0,0 +1,397 @@
+}
+
+void testContains() {
+    auto SpanSetLarge = geom::SpanSet::spanSetFromShape(4);
+    auto SpanSetSmall = geom::SpanSet::spanSetFromShape(1).shiftedBy(1, 1);
+    auto SpanSetSmallFar = geom::SpanSet::spanSetFromShape(1).shiftedBy(8, 8);
+    geom::Point2I pointIn(1, 1);
+    geom::Point2I pointOut(20, 20);
+
+    BOOST_CHECK(SpanSetLarge.contains(SpanSetSmall) == true);
+    BOOST_CHECK(SpanSetLarge.contains(SpanSetSmallFar) == false);
+    BOOST_CHECK(SpanSetLarge.contains(pointIn) == true);
+    BOOST_CHECK(SpanSetLarge.contains(pointOut) == false);
+}
+
+void testComputeCentroid() {
+    auto SpanSetShape = geom::SpanSet::spanSetFromShape(4).shiftedBy(2, 2);
+    auto center = SpanSetShape.computeCentroid();
+
+    BOOST_CHECK(center.getX() == 2);
+    BOOST_CHECK(center.getY() == 2);
+}
+
+void testComputeShape() {
+    auto SpanSetShape = geom::SpanSet::spanSetFromShape(1);
+    auto quad = SpanSetShape.computeShape();
+
+    BOOST_CHECK(quad.getIxx() == 0.4);
+    BOOST_CHECK(quad.getIyy() == 0.4);
+    BOOST_CHECK(quad.getIxy() == 0);
+}
+
+void testDilate() {
+    // Only need to test use of a stencil as it delegates to the other function overload
+    auto SpanSetPreDilate = geom::SpanSet::spanSetFromShape(1);
+    // This should dilate the shape by 1
+    auto SpanSetPostDilate = SpanSetPreDilate.dilate(1);
+
+    BOOST_CHECK(SpanSetPostDilate.getBBox().getMinX() == -2);
+    BOOST_CHECK(SpanSetPostDilate.getBBox().getMinY() == -2);
+}
+
+void testErode() {
+    // Only need to test use of a stencil as it delegates to the other function overload
+    auto SpanSetPreErode = geom::SpanSet::spanSetFromShape(2);
+    // This should erode the shape by 1
+    auto SpanSetPostErode = SpanSetPreErode.erode(1);
+
+    BOOST_CHECK(SpanSetPostErode.getBBox().getMinX() == -1);
+    BOOST_CHECK(SpanSetPostErode.getBBox().getMinY() == -1);
+}
+
+void testFlatten() {
+    // Test version without output array, as this simply delegates
+    // Create an array and initialize it to 9
+    ndarray::Array<int, 2, 2> input = ndarray::allocate(ndarray::makeVector(6, 6));
+    input.deep() = 9;
+
+    // overload a few pixels with the value 1
+    input[1][1] = 1;
+    input[1][2] = 1;
+    input[2][1] = 1;
+    input[2][2] = 1;
+    // Create a spanSet for those points
+    std::vector<geom::Span> spanVector = {geom::Span(0, 0, 1), geom::Span(1, 0, 1)};
+    geom::SpanSet SpnSt(spanVector);
+    auto flatArr = SpnSt.flatten(input, geom::Point2I(-1, -1));
+    for (auto const &  val : flatArr) {
+        BOOST_CHECK(val == 1);
+    }
+}
+
+void testUnflatten() {
+    // Test version without output array, as this simply delegates
+    ndarray::Array<int, 1, 1> input = ndarray::allocate(6);
+    input.deep() = 9;
+
+    // Create a SpanSet which has six points int it
+    std::vector<geom::Span> spanVector = {geom::Span(9, 2, 3), geom::Span(10, 3, 4), geom::Span(11, 2, 3)};
+    geom::SpanSet SpnSt(spanVector);
```

Comment:
Those guidelines say that obscure abbreviations should be avoided to aid in clarity. As dropping vowels is a common practice in programming, and the meaning is not really in question here, I think this is an appropriate name